### DISCUSSION TITLE
Help with Nextcloud AIO domain setup part

### DISCUSSION THREAD

**Discussion by alecuba** [nextcloud/all-in-one#4424] [Questions] [answered]
**Environment:**
**Ubuntu**
afa@nextcloud:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 23.10
Release:        23.10
Codename:       mantic

**Ubuntu Firewall output:**
afa@nextcloud:~$ sudo ufw status
Status: active

To                         Action      From
--                         ------      ----
22/tcp                     ALLOW       Anywhere
80/tcp                     ALLOW       Anywhere
443/tcp                    ALLOW       Anywhere
80,443/tcp                 ALLOW       Anywhere
3478/tcp                   ALLOW       Anywhere
3478/udp                   ALLOW       Anywhere
8080/tcp                   ALLOW       Anywhere
8443/tcp                   ALLOW       Anywhere
22/tcp (v6)                ALLOW       Anywhere (v6)
80/tcp (v6)                ALLOW       Anywhere (v6)
443/tcp (v6)               ALLOW       Anywhere (v6)
80,443/tcp (v6)            ALLOW       Anywhere (v6)
3478/tcp (v6)              ALLOW       Anywhere (v6)
3478/udp (v6)              ALLOW       Anywhere (v6)
8080/tcp (v6)              ALLOW       Anywhere (v6)
8443/tcp (v6)              ALLOW       Anywhere (v6)

Network **Router** with Public IP address setup with **port forwarding** to the Ubuntu server 192.168.0.71
443:443
8443:8443
3478:3478
8080:8080 -> created this to demonstrate that the domain resolves to the server I can see the setup page from the internet, also internally

**Docker setup with the following:**
sudo docker run -it\
--name nextcloud-aio-mastercontainer \ 
--restart always \ 
--publish 80:80 \ 
--publish 8080:8080 \ 
--publish 8443:8443 \ 
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \ 
--volume /var/run/docker.sock:/var/run/docker.sock:ro \ 
--env NEXTCLOUD_DATADIR="/mnt/buffalo_NAS/" \ 
nextcloud/all-in-one:latest

All works fine, the nexcloud storage gets created in the NAS share, but when trying to validate the domain it fails with the message:
"The domain is not reachable on Port 443 from within this container. Have you opened port 443/tcp in your router/firewall? If yes is the problem most likely that the router or firewall forbids local access to your domain. You can work around that by setting up a local DNS-server."

Domain is reachable from the internet an internally, I have even created an internal DNS to resolve the FQDN to the public IP address but still I am unable to validate the domain
While looking into the documentation I am quite confused with the [reverse proxy part](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md), if my fqdn is poitning to my server, is this reverse proxy still necessary? Isn't this part of the automation?

Any help is appreciated
Thanks

**Comment by szaimen** [comment]
Hi, I guess in your case you would need to [skip the domain validation](https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-skip-the-domain-validation) and then follow https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-can-i-access-nextcloud-locally

**Comment by alecuba** [comment]
Thanks @szaimen 
I tried as suggested, I even removed the containers and the existing configuration following [this article](https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-properly-reset-the-instance) 
Recreated all from the scratch with same config but added the suggested line to skip domain validation, this worked and all the containers came up but still I can NOT access the domain neither by fqdn or private IP

I do have a pi-hole, I created a custom DNS record for the FQDN as suggested in the documentation to point to the private IP
See nslookup from the Ubunto server renamed fqdn for obvious reasons.

afa@nextcloud:~$ nslookup
> nextcloud.<MYFQDN.COM>
Server:         127.0.0.53
Address:        127.0.0.53#53
Non-authoritative answer:
Name:   nextcloud.MYFQDN.COM
Address: **192.168.0.71**

I am pretty sure this is not a firewall issue because if I port forward my Router to another resource in 443 it does work

Now, I am pretty much confused on the[ reverse proxy part](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#reverse-proxy-documentation)
I have tried to understand where this needs to be configured
- Step 1 Configure the reverse proxy
Where and what is the file these changes to be implemented? 
I see few options and the recommended is Caddyfile, can you clarify where the caddy file is located? in what container?
I might be asking dumb questions but it is word for the sake of others trying to setup this nice tool

Thank you

**Reply by szaimen** [reply]
Aio handles the ssl stuff itself by default so you dont need to set up a reverse proxy yourself necessarily.

**Comment by alecuba** [comment]
Any recommendations on why this might be failing?
On the console I see these errors appearing often (Edited IP and FQDN):
{"level":"error","ts":1711203418.6103935,"logger":"tls.obtain","msg":"will retry","error":"[nextcloud.MYFQDN] Obtain: [nextcloud.MYFQDN] solving challenge: nextcloud.MYFQDN: [nextcloud.MYFQDN] authorization failed: HTTP 400 urn:ietf:params:acme:error:connection - <PUBLICIP>: Fetching http://nextcloud.MYFQDN/.well-known/acme-challenge/nct53f-xYTb1SBXGVgrGqPb9gYB62qQxH5nawTpKPck: Timeout during connect (likely firewall problem) (ca=https://acme-staging-v02.api.letsencrypt.org/directory)","attempt":2,"retrying_in":120,"elapsed":81.953446108,"max_duration":2592000}

**Reply by szaimen** [reply, marked as answer]
See https://github.com/nextcloud/all-in-one/discussions/2105
